test/TestServerProjectsLowLevel/dpg-customization/Generated/DPGClient.cs

```diff
@@ -0,0 +1,415 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace dpg_customization_LowLevel
+{
+    /// <summary> The DPG service client. </summary>
+    public partial class DPGClient
+    {
+        private const string AuthorizationHeader = "Fake-Subscription-Key";
+        private readonly AzureKeyCredential _keyCredential;
+        private readonly HttpPipeline _pipeline;
+        private readonly Uri _endpoint;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of DPGClient for mocking. </summary>
+        protected DPGClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of DPGClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public DPGClient(AzureKeyCredential credential, Uri endpoint = null, DPGClientOptions options = null)
```

Comment:
I just noticed that the endpoint is optional here - is this because of the swagger or generation?  We should not allow optional endpoints in client constructors.  I'm happy to handle this in a separate PR.

Reply:
I filed https://github.com/Azure/autorest.csharp/issues/2041 to track.